### DISCUSSION TITLE
2024.7.0

### DISCUSSION THREAD

**Discussion by AndrewBastin** [hoppscotch/hoppscotch#4225] [Announcements]
This is a major release with a couple of features (notably JUnit Reports for CLI, Infra Tokens in Self Hosted Editions, ability to duplicate collections), along with some bug fixes and some usability improvements in Self Hosted Editions. Read more about it here: https://hoppscotch.com/releases/release-notes-v2024-7-0

## What's Changed
* feat: Ability to toggle cookies to work in HTTP by @balub in https://github.com/hoppscotch/hoppscotch/pull/4194
* feat: duplicate team and user collections by @balub in https://github.com/hoppscotch/hoppscotch/pull/4207
* refactor(sh-admin): updated invite flow for the admin dashboard by @JoelJacobStephen in https://github.com/hoppscotch/hoppscotch/pull/4176
* feat(cli): add support for JUnit reporter by @jamesgeorge007 in https://github.com/hoppscotch/hoppscotch/pull/4189
* fix: auth bug when value is a secret environment by @nivedin in https://github.com/hoppscotch/hoppscotch/pull/4210
* HSB-462 feat: infra token module and sh apis by @mirarifhasan in https://github.com/hoppscotch/hoppscotch/pull/4191
* feat(sh-admin): introducing infra-tokens to admin dashboard by @JoelJacobStephen in https://github.com/hoppscotch/hoppscotch/pull/4202
* feat: duplicate REST/GraphQL collections by @jamesgeorge007 in https://github.com/hoppscotch/hoppscotch/pull/4211
* fix: infra token expriry check on guard by @mirarifhasan in https://github.com/hoppscotch/hoppscotch/pull/4219
* fix(sh-admin): resolved issue while rendering images by @JoelJacobStephen in https://github.com/hoppscotch/hoppscotch/pull/4221
* feat: hoppscotch-common & platform additions for ai experiments by @amk-dev in https://github.com/hoppscotch/hoppscotch/pull/4222

**Full Changelog**: https://github.com/hoppscotch/hoppscotch/compare/2024.6.1...2024.7.0

<hr /><em>This discussion was created from the release <a href='https://github.com/hoppscotch/hoppscotch/releases/tag/2024.7.0'>2024.7.0</a>.</em>

**Comment by baek-sang** [comment]
Are there any improvements to the desktop app in the works?

[fr](https://github.com/hoppscotch/hoppscotch/issues?q=is%3Aopen+is%3Aissue+label%3Adesktop+sort%3Areactions-%2B1-desc)

**Reply by AndrewBastin** [reply]
@baek-sang We are expecting some improvements to be up for the end of August release probably. Notably focused on two things.
1. Improvements with the login situation (mostly issues around Linux)
2. Improvements in app stability (again around Linux mostly)

We also have some work going on in an experimental basis around bringing SH support into the desktop app and work around it has started, but we might need a bit more time for that to cook properly as well.

**Reply by baek-sang** [reply]
Regretful desktop apps, self-hosted improvements aren't a priority

**Reply by AndrewBastin** [reply]
@baek-sang sorry that you feel that way, I am aiming to mostly close most of the stability issues this month (unless I go into a rabbit hole with one) and add desktop app specific improvements probably soon after. If you want it more faster, we welcome contributions! That will also allow me to focus on bringing SH support to the desktop app faster as well!

**Comment by yannlecann** [comment]
Is it possible to go directly from version 2023.12.5 to version 2024.7.0, or do you have to install intermediate versions

**Reply by AndrewBastin** [reply]
@yannlecann you should be able to update directly, make sure to run the migrations after updating and you should be fine.

Keeping a backup of the database won't hurt though!
